lib/lexer/lexer.dart

```diff
@@ -259,7 +260,21 @@ Token getNextToken(Lexer lexer) {
       case ']':
         return advanceWithToken(lexer, TokenType.TOKEN_RBRACKET);
       case ';':
-        return advanceWithToken(lexer, TokenType.TOKEN_SEMI);
+        final String value = lexer.currentChar;
+        const TokenType type = TokenType.TOKEN_SEMI;
+        advance(lexer);
+
+        if (lexer.currentChar == ';') {
+          int extras = 0;
+
+          while(lexer.currentChar == ';')
+            advance(lexer);
+            extras++;
+
+          print(Warning('Unnecessary trailing semicolons', lexer.program, 'remove these extra semicolons', lexer.lineNum, extras, lexer.currentIndex, lexer.fileName));
```

Comment:
- This is likely to get used a lot. Refactor it into a `warn(Lexer lexer, String message, String help)`.
- I don't feel like printing the warning is the responsibility of the lexer. Maybe have a global function that reports warnings for now.

Reply:
I think the warning and help messages should be constants in the Warning class, or better yet we had a named constructor like `Warning.extraSemicolons`, or even a subclass `ExtraSemicolonsWarning`.
When we have other kinds of warnings, we won't want these messages to be scattered all over the place.

Reply:
Actually we cannot directly pass the lexer as if we expand the warning system then its more likely to be used in the runtime and parser to so passing the lexer is not the best choice. CC @Calamity210

Reply:
I meant `warn` to be in lexer.dart. It's specific to the lexer, not part of Warning. Parser and Runtime should have their own warn methods too when we add warnings in them.

Reply:
I see what you mean. So make methods smth like WarningLexer, WarningParser, WarningRuntime and it will make the code prettier!